libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1455,10 +1461,20 @@ void IRGeneratorForStatements::declareAssign(IRVariable const& _lhs, IRVariable
 			_rhs.commaSeparatedList() <<
 			")\n";
 }
-void IRGeneratorForStatements::declare(IRVariable const& _var)
+
+void IRGeneratorForStatements::allocateMemoryAndDeclareVariable(VariableDeclaration const& _var)
```

Comment:
It's not so much about allocating memory but rather about assigning the initial value. Can you just call and assign `zeroValue` for all types?

Reply:
I think if you use an IRVariable of the correct type, you also don't have to bother about mentioning `"mpos"`

Reply:
I can, but that will lead to redundant assignments for every value type. Do we trust the optimizer? ;)

Reply:
the optimizer is a requirement for this code to be efficient - that's the philosophy.